packages/core/frontend/sirius-components-core/src/selection/useSelectionUrlSearchParams.ts

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { SetURLSearchParams } from 'react-router-dom';
```

Comment:
- The entire contribution is related to the EditProjectView in Sirius Web, no need to contribute anything in `sirius-components-core`
- This file is named `useXxx`  like a React hook but its content is not a hook at all
- During the rendering of `EditProjectView`, you are calling `updateSelectionBasedOnUrlSearchParamsIfNeeded` which has a side effect (it calls setSelection). While rendering a React component, you cannot use side effects